ignite/metrics/rouge.py

```diff
@@ -0,0 +1,438 @@
+        return "Rouge-L"
+
+
+class Rouge(Metric):
+    r"""Calculates the Rouge score for multiples Rouge-N and Rouge-L metrics.
+
+    More details can be found in `Lin 2004`__.
+
+    __ https://www.aclweb.org/anthology/W04-1013.pdf
+
+    - ``update`` must receive output of the form ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+    - `y_pred` must be a sequence of tokens.
+    - `y` must be a list of sequence of tokens.
+
+    Args:
+        metrics: names of metrics computed. Valid names are "Rouge-L" and "Rouge-n" where n is an integer 1 <= n <= 9.
+        multiref: reduces scores for multi references. Valid values are "best" and "average" (default: "average").
+        alpha: controls the importance between recall and precision (alpha -> 0: recall is more important, alpha -> 1:
+           precision is more important)
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+        device: specifies which device updates are accumulated on. Setting the metric's
+            device to be the same as your ``update`` arguments ensures the ``update`` method is non-blocking. By
+            default, CPU.
+
+    Example:
+
+    .. code-block:: python
+
+        from ignite.metrics import Rouge
+
+        m = Rouge(metrics=["Rouge-L", "Rouge-2"], multiref="best")
+
+        candidate = "the cat is not there".split()
+        references = [
+            "the cat is on the mat".split(),
+            "there is a cat on the mat".split()
+        ]
+
+        m.update((candidate, references))
+
+        m.compute()
+        >>> {'Rouge-L-P': 0.6, 'Rouge-L-R': 0.5, 'Rouge-L-F': 0.5, 'Rouge-2-P': 0.5, 'Rouge-2-R': 0.4, 'Rouge-2-F': 0.4}
+
+        .. versionadded:: 0.5.0
+        """
+
+    def __init__(
+        self,
+        metrics: Optional[Sequence[str]] = None,
+        multiref: str = "average",
+        alpha: float = 0,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ):
+        if metrics is None or len(metrics) == 0:
+            metrics = ["Rouge-1", "Rouge-2", "Rouge-4", "Rouge-L"]
+        self.internal_metrics: List[_BaseRouge] = []
+        for m in metrics:
+            if m == "Rouge-L":
+                self.internal_metrics.append(RougeL(multiref, alpha, output_transform, device))
+            else:
+                match = re.match("Rouge-(?P<ngram>[1-9])", m)
+                if not match:
+                    raise ValueError("Rouge metric name must be 'Rouge-L' or 'Rouge-N' for N in [1 ... 9]")
+                n = int(match.group("ngram"))
+                self.internal_metrics.append(RougeN(n, multiref, alpha, output_transform, device))
```

Comment:
Nit: maybe we could use input metrics as `Sequence[Union[str, int]]` and receive `["L", 1, 2, 4]` and simplify regexp etc ?

Reply:
```
m = Rouge(metrics=["Rouge-1", "Rouge-2", "Rouge-4", "Rouge-L"], multiref="best", alpha=0.5)
```
would become
```
m = Rouge(metrics=[1, 2, 4, "L"], multiref="best", alpha=0.5)
```
I confess that is more compact but less clear to me. You choose :)

Reply:
Class `Rouge` is a group of Rouge-* variants. Maybe, it would make also sense to expose others: RougeN, RougeL in `ignite/metrics/__init__.py` ?

I'd code it even:
```python
m = Rouge(variants=[1, 2, 4, "L"], multiref="best", alpha=0.5)
```
But I have not strong opinion about that, just make things "simplier" probably